plugins/datadog-entity-sync-node/src/services/DatadogServiceFromEntitySync.ts

```diff
@@ -146,20 +150,38 @@ export class DatadogServiceFromEntitySync<
       });
       try {
         const entityTitle = entity.metadata.title ?? entity.metadata.name;
-        const service = JSON.stringify(this.serialize(entity, preload));
+        const service = yaml.dump(this.serialize(entity, preload));
         assert(
           service,
           `The entity ${entityTitle} was unable to be processed.`,
         );
 
+        const apiKey = this.#clients.config.getString(
+          'datadog.integration.apiKey',
+        );
+        const appKey = this.#clients.config.getString(
+          'datadog.integration.appKey',
+        );
+        const site =
+          this.#clients.config.getOptionalString('datadog.integration.site') ??
+          'datadoghq.com';
+        const baseUrl = `https://api.${site}`;
+
         if (!this.#enabled || dryRun) {
           logger.info(
             `The entity ${entityTitle} was not synced due to the sync being disabled.`,
           );
 
           yield service;
         } else {
-          yield await this.#clients.datadog.upsertCatalogEntity({
+          yield await fetch(`${baseUrl}/api/v2/catalog/entity`, {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'text/plain',
+              Accept: 'application/json',
+              'DD-API-KEY': apiKey,
+              'DD-APPLICATION-KEY': appKey,
+            },
```

Comment:
https://github.com/DataDog/datadog-backstage-plugins/blob/d446d6a6031d7913e77faed04a26b08d483de62c/plugins/datadog-entity-sync-node/src/datadogSoftwareCatalogApi.ts

shouldn't this be using the v3 enpoint now? ANd the v3 client? Why are we replacing the fetch call here rather than using the datadog client?